### PR TITLE
Integrate Signal API alerting

### PR DESCRIPTION
**Pull request by cooperose**
Added Signal as an alerting option following code structure of other alerting platforms. All changes are additions and shouldn’t break any current installs. 

I added a couple of networking notes to the rsync setup document. Having separate vlans with a zero trust setup had me digging to figure out how Teslausb was checking if the archive was available. Added clarification for ICMP usage as well as rsync port.

### CODE REVIEW

File: setup/pi/configure.sh [hunk 1]
```diff
@@ -251,6 +251,25 @@ function install_matrix_packages () {
...
+function check_signal_configuration () {
+  if [ "${SIGNAL_ENABLED:-false}" = "true" ]
+  then
+    if [ -z "${SIGNAL_URL+x}" ] || [  "${SIGNALTO_NUM+x}" = "country_code_and_number_configured_with_signal" ] || [  "${SIGNAL_FROM_NUM+x}" = "country_code_and_number_configured_with_signal"  ]
+    then
+      log_progress "STOP: You're trying to setup Signal but didn't provide a URL."
+      log_progress "Define the variables like this:"
+      log_progress "export URL_PORT=put_protocol_ip/hostname_portnumber"
+      log_progress "export TO_NUM=put_phone_number_associated_with_signal_including_country_code"
+      log_progress "export FROM_NUM=put_phone_number_associated_with_signal_including_country_code_to_send_to"
```
Comment (marcone) [inline setup/pi/configure.sh]: This is not what these variables are actually called

**Review by marcone — CHANGES_REQUESTED** [review]
Thanks. Looks OK except for some minor things.

File: setup/pi/configure.sh (hunk 1 above)
Comment (cooperose) [reply setup/pi/configure.sh]: great catch! my bad. Those variables have been updated.

**Comment by marcone** [comment]
Please fix the shellcheck errors (should just be a matter of adjusting/adding quotes)

**Comment by marcone** [comment]
Closing due to no response. Feel free to resubmit once the shellcheck issues have been addressed.

**Pull request closed without merging**